java/src/com/ibm/streamsx/topology/internal/context/remote/RemoteContexts.java

```diff
@@ -0,0 +1,38 @@
+package com.ibm.streamsx.topology.internal.context.remote;
+
+import static com.ibm.streamsx.topology.internal.gson.GsonUtilities.jstring;
+import static com.ibm.streamsx.topology.internal.gson.GsonUtilities.object;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.JsonObject;
+import com.ibm.streamsx.topology.context.remote.RemoteContext;
+
+public class RemoteContexts {
+    /**
+     * Write the results output to the results file.
+     * @param submission The JsonObject representation of the application.
+     * @return The Future<> result of the submission.
+     */
+    public static void writeResultsToFile(JsonObject submission) {
+        // Get the results file location.
+        String resultsFile = jstring(submission, RemoteContext.SUBMISSION_RESULTS_FILE);
+        if(resultsFile == null)
+            return;
+    
+        // Write to the file and close the file.
+        List<String> lines = new ArrayList<>();
```

Comment:
Why do you need the `List`?

Just write `results_json.toString()` directly, though ensure it is written using UTF-8 encoding.

Something like.

`Files.writer(Paths.get(resultsFile), resullts_json.toString().getBytes(StandardCharsets.UTF_8))`

Otherwise if you want to use a list then `Collections.singletonList` is cleaner.